### PR TITLE
THRIFT-3784: fix and tests

### PR DESCRIPTION
**Pull request by csosborn**
This patch fixes [THRIFT-3784](https://issues.apache.org/jira/browse/THRIFT-3784), a bug in the thrift-maven-plugin.

### CODE REVIEW

File: contrib/thrift-maven-plugin/src/main/java/org/apache/thrift/maven/AbstractThriftMojo.java [hunk 1]
```diff
@@ -319,15 +331,6 @@ public boolean accept(File dir, String name) {
...
     }
 
-    ImmutableSet<File> findThriftFilesInDirectories(Iterable<File> directories) throws IOException {
-        checkNotNull(directories);
-        Set<File> thriftFiles = newHashSet();
-        for (File directory : directories) {
-            thriftFiles.addAll(findThriftFilesInDirectory(directory));
-        }
-        return ImmutableSet.copyOf(thriftFiles);
-    }
```
Comment (csosborn) [inline contrib/thrift-maven-plugin/src/main/java/org/apache/thrift/maven/AbstractThriftMojo.java]: This code was unused.

**Comment by csosborn** [comment]
What's the protocol here? It does not look like the build failures have anything to do with my changes.

**Comment by csosborn** [comment]
Just checking in here.. This request has been open for about two months with no response. 

Is the project still accepting patches? It doesn't look like anything has been merged in a few months, and the CI system appears unhealthy.

**Comment by jeking3** [comment]
The project is still accepting patches, however as you can see we have over 100 pull requests (I have 3 in the queue) but we haven't been able to stabilize the apache build job for quite some time, leading to a backlog.  Only a few people have access to the apache build job and until we can get builds to pass the project lead decided (correctly, in my opinion) not to merge any more code.

**Pull request closed without merging**